### PR TITLE
(PCP-693) Update client to be compatible with pcp-broker 1.0

### PR DESCRIPTION
**Pull request by MikaelSmith**

### CODE REVIEW

File: src/puppetlabs/pcp/client.clj [hunk 1]
```diff
@@ -325,6 +325,14 @@
     (.start client)
     client))
 
+(defn ^:private append-client-type
+  "Append client type to a ws connection Uri, accounting for possible trailing
+   slash."
+  [url type]
+  (if (= (last url) \/)
```
Comment (mruzicka) [inline src/puppetlabs/pcp/client.clj]: I'd probably use java's `String#endsWith` to avoid creation & iteration through the collection of characters.

Comment (MikaelSmith) [inline src/puppetlabs/pcp/client.clj]: I wonder if it actually iterates, or optimizes for known classes.

Comment (mruzicka) [inline src/puppetlabs/pcp/client.clj]: I'm not sure if there is any optimization elsewhere, but this implementation does iterate: https://github.com/clojure/clojure/blob/clojure-1.8.0/src/clj/clojure/core.clj#L262-L270

Comment (MikaelSmith) [inline src/puppetlabs/pcp/client.clj]: We depend on clojure 1.8, so I think I'll use https://clojure.github.io/clojure/branch-master/clojure.string-api.html#clojure.string/ends-with?.

Comment (MikaelSmith) [inline src/puppetlabs/pcp/client.clj]: Tested it.
```
user=> (time (every? #(clojure.string/ends-with? % "/") (repeat 10000000 "wss://broker.example.com/foo/")))
"Elapsed time: 436.909986 msecs"
true
user=> (time (every? #(= (last %) \/) (repeat 10000000 "wss://broker.example.com/foo/")))
"Elapsed time: 11243.748476 msecs"
true
```

**Comment by MikaelSmith** [comment]
Updated.

**Comment by MikaelSmith** [comment]
The logging test seems to be somewhat unreliable. Not sure what to do about it right now.

File: src/puppetlabs/pcp/client.clj (hunk 1 above)
Comment (mruzicka) [inline src/puppetlabs/pcp/client.clj]: 👍

**Pull request merged**